### ISSUE TITLE
Default security on Reminder Links block should be changed

### ISSUE THREAD

**Issue by austinspooner** [SparkDevNetwork/Rock#5567]
### Please go through all the tasks below

- [X] Check this box only after you have successfully completed both the above tasks

### Please provide a brief description of the problem. Please do not forget to attach the relevant screenshots from your side.

The default security on the Reminder Links block only has view access enabled for the Staff Workers and Staff Like Workers Security roles by default. 

### Expected Behavior

The default behavior should be that the Staff Workers and Staff Like Workers security roles have "edit" rights to the Reminder Links block by default. Without edit rights, both those security roles cannot create reminders. Out of the box, I think those roles should be able to create reminders. 

### Actual Behavior

Only Rock Administrators can create new reminders out of the box without modifying security.

### Steps to Reproduce

- Go to the Rock home page
- Click Alt-B for block configuration
- Choose the security padlock for the Reminder Links block
- View security 
- Notice that the only group with Edit access (edit in this case enables creating new reminders) is Rock Administrators. 

### Rock Version

15.1 & 16.0

### Client Culture Setting

en-us

**Comment by shauncummings** [comment]
Hey @austinspooner.  I am unable to duplicate this issue as described.  As you can see from the screenshot below, when I log into the public demo site as Cindy Decker (cdecker/password) you can definitely see the "Add Reminder" option is available to her

<img width="749" alt="image" src="https://github.com/SparkDevNetwork/Rock/assets/46500397/3a90a766-687b-4a64-a948-a47b9bec51d9">

I verified that she can in fact add a reminder, view it, and react to it.  Cindy is not an admin, which you can verify in her [security settings](https://www.rockrmsdemo.com/person/5/security).

There are a few things going on here that may be contributing to your confusion.  There are various checks for conditions to enable the "Add Reminder" option on any given page, and for performance reasons these checks are performed after the page is initially rendered by the browser.  To see the "Add Reminder" option, the following conditions must be met:

1. The page you are viewing must have a context entity (by default, the `/person/[id]` and `/group/[id]` routes will suffice for testing this).
2. A reminder type must exist for the type of the context entity type (again, by default we provide these for Person and Group entities).
3. The logged in user must have VIEW permission to the reminder type.  (All users have this by default, but you may restrict it.)
4. JavaScript must properly execute on the page after the DOM is loaded.  (If an error occurs on the page prior to this event firing, it could be preventing the "Add Reminder" button from showing up.)
5. A REST request to `api/ReminderTypes/ReminderTypesExistForEntityType` must complete successfully for the logged in user.  (This should be fine under default conditions, but again, there are security settings you can alter which would limit access to this endpoint.)

I'm going to go ahead and close this issue since I cannot duplicate it according to the existing description.  If you find another problem with the hints provided above, I recommend opening a new issue.  If you think there's something I missed here, you can re-open this one.
